semver_test.go

```diff
@@ -97,6 +97,27 @@ func TestNewSemverRequirements_tilde(t *testing.T) {
 	check("~1.2.3-beta.2", []string{"1.2.3-beta.2", "1.2.3", "1.2.4", "1.2.3-beta.3"}, []string{"1.2.2", "1.0.0", "1.2.4-beta.2", "1.3.0"})
 }
 
+func TestNewSemverRequirements_bad(t *testing.T) {
```

Comment:
Suffix this with `_invalid` instead of `_bad` and set it up as a "table test" so it's easier to add additional cases.

Something like:
```go
check := func(semverStr string) {
  _,err = // ...
  if err == nil {
    t.Errorf("check '%s': got nil, expected err", semverStr)
  }
}

check(">=")
check(">")
check("~")
// maybe add some other invalid ones too (like invalid characters or unicode cases) -- see what happens
check("💩")
```

Reply:
Hah yea me being a bit lazy again, good call

Reply:
No problem. TBH I feel I should rewrite all the tests in the "got; want" style and using the built in table test support (using `t.Run`) but I think that would be better done after the PR to keep it to the point.